app/javascript/components/public_health/AdvancedFilter.js

```diff
@@ -567,58 +613,114 @@ class AdvancedFilter extends React.Component {
                 </Row>
               </Form.Group>
             )}
-            {filterOption?.type === 'date' && dateOption != 'within' && (
-              <Form.Group className="py-0 my-0">
-                <DateInput
-                  date={value}
-                  onChange={date => {
-                    this.changeValue(index, date);
-                  }}
-                  placement="bottom"
-                  customClass="form-control-md"
-                  minDate={'1900-01-01'}
-                  maxDate={moment()
-                    .add(2, 'years')
-                    .format('YYYY-MM-DD')}
-                />
-              </Form.Group>
-            )}
-            {filterOption?.type === 'date' && dateOption === 'within' && (
-              <Form.Group className="py-0 my-0">
-                <Row>
-                  <Col className="pr-0">
-                    <DateInput
-                      date={value.start}
-                      onChange={date => {
-                        this.changeValue(index, { start: date, end: value.end });
-                      }}
-                      placement="bottom"
-                      customClass="form-control-md"
-                      minDate={'1900-01-01'}
-                      maxDate={moment()
-                        .add(2, 'years')
-                        .format('YYYY-MM-DD')}
-                    />
-                  </Col>
-                  <Col className="py-0 px-0 text-center my-auto" md="2">
-                    <b>TO</b>
+            {filterOption?.type === 'date' && (
+              <Row>
+                <Col className="py-0" md="6">
+                  {this.renderDateOptions(dateOption, index)}
+                </Col>
+                {dateOption != 'within' && (
+                  <Col className="py-0">
+                    <Form.Group className="py-0 my-0">
+                      <DateInput
```

Comment:
I see this entire `DateInput` duplicated in a lot of places where the only difference seems to be the params to the `changeValue()` method. Could you pull out into function that just takes that as an argument?

Reply:
this actually gets pretty wonky with value being an object vs just a date (among other things). a new render would need 4 different arguments and would have a decent amount of logic. my vote is to leave as is because it's cleaner to understand